drag-drop-swipe-recyclerview/src/main/java/com/ernestoyaquello/dragdropswiperecyclerview/DragDropSwipeAdapter.kt

```diff
@@ -531,6 +538,7 @@ abstract class DragDropSwipeAdapter<T, U : DragDropSwipeAdapter.ViewHolder>(
             canvasOver: Canvas?,
             isUserControlled: Boolean) {
 
+        if (viewHolder.adapterPosition == NO_POSITION) return
```

Comment:
Is this check really necessary? Two lines below, this is already checked with the following line:

```kotlin
val item = if (currentAdapterPosition != -1) dataSet[currentAdapterPosition] else null
```

Even though, for consistency, now that line should be changed to this:

```kotlin
val item = if (currentAdapterPosition != NO_POSITION) dataSet[currentAdapterPosition] else null
```

Reply:
Yes. In fact, this case was already discussed in the documentation of the method `onIsSwiping`:

> It may be null if this method is being called because the item layout is still being animated by the system but the item itself has already been removed from the data set.